src/init.cpp

```diff
@@ -1697,6 +1710,19 @@ bool AppInitMain(NodeContext& node)
         GetBlockFilterIndex(filter_type)->Start();
     }
 
+    if (nLocalServices & NODE_COMPACT_FILTERS) {
+        const BlockFilterIndex* const basic_filter_index =
+            GetBlockFilterIndex(BlockFilterType::BASIC);
+        if (!basic_filter_index) {
+            error("NODE_COMPACT_FILTERS is signaled, but filter index is not available");
+            return false;
+        }
+        if (!basic_filter_index->IsSynced()) {
+            InitError(strprintf(_("Cannot enable -peercfilters until basic block filter index is in sync. Please disable and reenable once filters have been indexed.").translated));
```

Comment:
I ran into this error when I had to restart a still-syncing node. So while I can add the `-peercfilters` flag when starting the node for the first time and syncing from scratch, I have to remove it when restarting a syncing node (otherwise it'd refuse to start with this error). I think it's better to either always allow the `-peercfilters` flag or only allow it once the filter index is ready. Former being preferred, since if combined with https://github.com/bitcoin/bitcoin/pull/16442/files#r363580980, it allows for everything to work with the `-peercfilters` flag only.

Other than that: amazing work!

Reply:
# # # # # # #